### PR TITLE
Warn while protocol plugin register failed

### PR DESCRIPTION
**Pull request by medcl**
I just pick up code and try to finish my Stmp protocol,but didn't found the plugin interface has already changed,it took me some time to debug out,and i think it is better to print a warn message if the protocol plugin is not successful registered.

### CODE REVIEW

File: packetbeat/protos/protos.go [hunk 1]
```diff
@@ -231,11 +231,17 @@ func (protocols ProtocolsStruct) BpfFilter(with_vlans bool, with_icmp bool) stri
 
 func (protos ProtocolsStruct) Register(proto Protocol, plugin ProtocolPlugin) {
 	protos.all[proto] = plugin
+	var success bool = false
```
Comment (andrewkroh) [inline packetbeat/protos/protos.go]: This logic would be well suited for a [type switch](https://golang.org/doc/effective_go.html#type_switch) where the default case logs the warning.

Comment (medcl) [inline packetbeat/protos/protos.go]: ok, let me try

**Comment by andrewkroh** [comment]
For your protocol plugin you can statically validate that it implements a given interface by using something like [this line](https://github.com/elastic/beats/blob/master/packetbeat/protos/dns/dns_tcp_test.go#L26). This causes the compiler to validate that the type implements the specified interface.

File: packetbeat/protos/protos.go (hunk 1 above)
Comment (medcl) [inline packetbeat/protos/protos.go]: @andrewkroh seems plugin like memcache implement both tcp and udp protocol, so i think it is not a good idea to use type switch here,what you think?

**Comment by urso** [comment]
:+1: for this change. It's all to easy to break plugin registration. 

How about making this check more reliable by having methods on ProtocolPlugin interface telling the supported protocols. e.g.

    type ProtocolPlugin interface {
        IsTCPPlugin() bool
        IsUDPPlugin() bool
    }

If e.g. UDP Protocol plugin type cast fails, but protocol plugin states it is UDP we print an error.

Some protocols (e.g. memcache or dns) support TCP and UDP. Without additional checks we can not tell if registration failed in case just one of interface for TCP/UDP plugins changed.

File: packetbeat/protos/protos.go (hunk 1 above)
Comment (andrewkroh) [inline packetbeat/protos/protos.go]: @medcl You're right. Ignore my comment, I read that block as if it was an if/else which it is clearly not.

**Comment by tsg** [comment]
Merging this one, the interface change should be done in a separate PR.

**Pull request merged**